zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/FileSystemStorage.java

```diff
@@ -60,16 +63,42 @@
   public FileSystemStorage(ZeppelinConfiguration zConf, String path) throws IOException {
     this.zConf = zConf;
     this.hadoopConf = new Configuration();
+    URI zepConfigURI;
+    URI defaultFSURI;
+
+    zepConfigURI = getURI(path, "Failed to get URI path");
     // disable checksum for local file system. because interpreter.json may be updated by
     // non-hadoop filesystem api
     // disable caching for file:// scheme to avoid getting LocalFS which does CRC checks
+
     this.hadoopConf.setBoolean("fs.file.impl.disable.cache", true);
-    this.hadoopConf.set("fs.file.impl", RawLocalFileSystem.class.getName());
+    String defaultFS = this.hadoopConf.get(FS_DEFAULTFS);
+    defaultFSURI = getURI(defaultFS, "Failed to get defaultFS URI");
+
+    if (isS3AFileSystem(defaultFSURI, zepConfigURI)) {
+      this.hadoopConf.set("fs.s3a.impl", S3AFileSystem.class.getName());
```

Comment:
Why we have to set it explicitly? Hadoop can not detect the s3a filesystem ?

Reply:
I have tried without setting s3a filesystem but it was not working, so i had set it explicitly; still i will give one more try without setting s3a filesystem.

Reply:
I doubt why hadoop doesn't support it, ideally hadoop should support this otherwise any downstream project has to set s3 filesystem explicitly which is annoying.

Reply:
if we remove explicitly setting of fs.s3a.impl as S3AFileSystem, then it throws the class cast exception as it sets fs.file.impl as RawLocalFileSystem. So do we also need to remove the explicitly setting of fs.file.impl?

Reply:
I am suggesting to remove it. I just want to confirm why hadoop didn't detect the right FileSystem implementation. As my understanding, hadoop should be able to detect the right FileSystem implementation based on the filesystem schema (e.g. file://, hdfs:// and etc). So could you help investigate why it doesn't work for s3. Thanks

Reply:
After removing 'fs.file.impl' it works with all the filesystem schema (file:///, hdfs:///, s3a://)

Reply:
@zjffdu : If anything is not missing then can I merge?

Reply:
I think you need to update this ticket and PR description. There's no issue to connect to s3a. Only the compilation issue with hadoop3 is fixed in this PR IIUC.

Reply:
@zjffdu I have updated ticket and PR description.